app/src/main/java/org/schabi/newpipe/player/playback/MediaSourceManager.java

```diff
@@ -180,13 +198,15 @@ private MediaSourceManager(@NonNull final PlaybackListener listener,
      */
     public void dispose() {
         if (DEBUG) {
-            Log.d(TAG, "close() called.");
+            Log.d(TAG, "dispose() called.");
         }
 
         debouncedSignal.onComplete();
         debouncedLoader.dispose();
 
         playQueueReactor.cancel();
+        //TODO: Why not clear here?
+        //TODO: Also why not clear loadingItems here?
```

Comment:
Good question. I think this is a cood catch. I'd clear loadingItems as the end. However, I do not see what you'd like to clear additionally.